content-security-policy/reporting/report-clips-sample.https.html

```diff
@@ -9,27 +9,73 @@
 </head>
 <body>
   <script>
+  const trimmedSampleLength = 40;
+  const evalSample = "evil = '1234567890123456789012345678901234567890;'";
+  const trimmedEvalSample = evalSample.substring(0, trimmedSampleLength);
   promise_test(t => {
     let evil = false;
     assert_throws_js(EvalError, _ => {
-      eval("evil = '1234567890123456789012345678901234567890';");
+      eval(evalSample);
     });
     assert_false(evil);
     return waitUntilCSPEventForTrustedTypes(t).then(t.step_func_done(e => {
-      assert_equals(e.sample, "eval|evil = '12345678901234567890123456789012");
+      assert_equals(e.sample, `eval|${trimmedEvalSample}`);
     }));
-  }, "Unsafe eval violation sample is clipped to 40 characters.");
+  }, `Unsafe eval violation sample is clipped to ${trimmedSampleLength} characters.`);
+
+  promise_test(t => {
+    let evil = false;
+    assert_throws_js(EvalError, _ => {
+      eval?.(evalSample);
+    });
+    assert_false(evil);
+    return waitUntilCSPEventForTrustedTypes(t).then(t.step_func_done(e => {
+      assert_equals(e.sample, `eval|${trimmedEvalSample}`);
+    }));
+  }, `Unsafe indirect eval violation sample is clipped to ${trimmedSampleLength} characters.`);
+
+  const functionBody = "return '1234567890123456789012345678901234567890';";
+  const sampleWithoutFunctionPrefix = `(a,b\n) {\nreturn ${functionBody}\n}`;
```

Comment:
@lukewarlow  I hadn't checked the change again after approving, but sampleWithoutFunctionPrefix should not have the `return ` statement, since it's already in functionBody.

Reply:
Apologies I should have spotted that! (None of the browsers currently pass this test so didn't actually run it to double check).